src/tracing/errors.test.ts

```diff
@@ -1,12 +1,68 @@
-import { ErrorEvent } from '@sentry/types'
+import { ErrorEvent, Event } from '@sentry/types'
 
-import { filterKnownErrors } from './errors'
+import { beforeSend, filterKnownErrors } from './errors'
 
 Object.defineProperty(window.performance, 'getEntriesByType', {
   writable: true,
   value: jest.fn(),
 })
 
+describe('beforeSend', () => {
+  const ERROR = {} as ErrorEvent
+
+  describe('chunkResponseStatus', () => {
+    afterEach(() => {
+      jest.restoreAllMocks()
+    })
+
+    it('handles when matching JS file not found', () => {
+      jest.spyOn(window.performance, 'getEntriesByType').mockReturnValue([])
+      const originalException = new Error(
+        'Loading chunk 20 failed. (error: https://app.uniswap.org/static/js/20.d55382e0.chunk.js)'
+      )
+      expect((beforeSend(ERROR, { originalException }) as Event).tags).toBeUndefined()
+    })
+
+    it('handles when matching CSS file not found', () => {
+      jest.spyOn(window.performance, 'getEntriesByType').mockReturnValue([])
+      const originalException = new Error(
+        'Loading chunk 20 failed. (error: https://app.uniswap.org/static/js/20.d55382e0.chunk.js)'
+      )
+      expect((beforeSend(ERROR, { originalException }) as Event).tags).toBeUndefined()
+    })
+
+    it('handles when performance is undefined', () => {
+      window.performance = undefined as any
+      const originalException = new Error('Loading CSS chunk 12 failed. (./static/css/12.d5b3cfe3.chunk.css)')
+      expect((beforeSend(ERROR, { originalException }) as Event).tags).toBeUndefined()
+    })
+
+    it('adds status for a matching JS file', () => {
+      jest.spyOn(window.performance, 'getEntriesByType').mockReturnValue([
+        {
+          name: 'https://app.uniswap.org/static/js/20.d55382e0.chunk.js',
+          responseStatus: 499,
+        } as PerformanceEntry,
+      ])
+      const originalException = new Error(
+        'Loading chunk 20 failed. (error: https://app.uniswap.org/static/js/20.d55382e0.chunk.js)'
+      )
+      expect((beforeSend(ERROR, { originalException }) as Event).tags?.chunkResponseStatus).toBe(499)
+    })
+
+    it('adds status for a matching CSS file', () => {
+      jest.spyOn(window.performance, 'getEntriesByType').mockReturnValue([
+        {
+          name: 'https://app.uniswap.org/static/css/12.d5b3cfe3.chunk.css',
+          responseStatus: 200,
+        } as PerformanceEntry,
+      ])
+      const originalException = new Error('Loading CSS chunk 12 failed. (./static/css/12.d5b3cfe3.chunk.css)')
+      expect((beforeSend(ERROR, { originalException }) as Event).tags?.chunkResponseStatus).toBe(200)
+    })
+  })
+})
+
 describe('filterKnownErrors', () => {
```

Comment:
Isn't this exported solely for testing?
Should it be tested through beforeSend? Alternatively, could you export your chunk helper method for isolated testing?

Reply:
probably, i can change that all in a separate pr

Reply:
tbh we could probably lint for this too

Reply:
https://github.com/Uniswap/interface/pull/6513